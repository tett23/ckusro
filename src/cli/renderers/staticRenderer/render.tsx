import React from 'react';
import { renderToString } from 'react-dom/server';
import { ServerStyleSheet, StyleSheetManager } from 'styled-components';
import App, { Props } from './assets/components';

export type RenderResult = {
  html: string;
  styles: string;
};

export default function render(props: Props): RenderResult {
  const sheet = new ServerStyleSheet();
  const html = renderToString(
    <StyleSheetManager sheet={sheet.instance}>
      <App {...props} />
    </StyleSheetManager>,
  );
  const styles = sheet.getStyleTags();

  return {
    html,
    styles,
  };
}
