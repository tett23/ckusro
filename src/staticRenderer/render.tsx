import React from 'react';
import ReactDOMServer from 'react-dom/server';
import App, { Props } from './assets/components';

export default function render(props: Props): string {
  return ReactDOMServer.renderToString(<App {...props} />);
}
