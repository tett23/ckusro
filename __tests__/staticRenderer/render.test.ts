import render from '../../src/staticRenderer/render';
import {
  buildFile,
  buildGlobalState,
  buildLoaderContext,
} from '../__fixtures__';

describe(render, () => {
  it('renders correctly', () => {
    const file = buildFile();
    const contexts = [buildLoaderContext({ name: file.namespace })];
    const globalState = buildGlobalState({ files: [file] });
    const actual = render({
      globalState,
      markdown: {
        currentFileId: file.id,
        files: [file],
      },
    });

    expect(actual).toMatchSnapshot();
  });
});
