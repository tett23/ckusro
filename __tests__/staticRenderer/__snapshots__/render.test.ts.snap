// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `"<main data-reactroot=\\"\\"><nav><ul></ul></nav><div><nav><nav><ul><li><a href=\\"..\\">test</a></li><li><a href=\\"./\\">foo.md</a></li></ul></nav></nav><div><section><p><span data-reactroot=\\"\\">bar.md</span></p></section><section><pre>[[bar.md]]</pre></section></div></div></main>"`;
